src/auth_end.ejs

```diff
@@ -7,12 +7,13 @@
     <script>
       const params = new URLSearchParams(window.location.search);
       const result = params.get('code');
-      const { authId, method } = JSON.parse(params.get('state'));
+      const { authId, method, redirectUrl } = JSON.parse(params.get('state'));
+      const hostRedirectUrl = redirectUrl || `msteams://teams.microsoft.com/l/auth-callback?authId=${encodeURIComponent(authId)}&result={result}`;
       if (method === 'deeplink') {
-        window.location.href = `msteams://teams.microsoft.com/l/auth-callback?authId=${authId}&result=${result}`
+        window.location.href = hostRedirectUrl.replace('{result}', encodeURIComponent(result));
```

Comment:
The authId and result weren't being encoded before. Did you intentionally add that logic?

Reply:
yes, I had to encode the result and noticed the other params weren't being encoded. Nothing really broke because the sample just provided URL-safe values like `1` and `1234` but for completeness sake I just added it for all of them.